components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LoggingResource.java

```diff
@@ -108,65 +117,68 @@ public boolean invoke(MessageContext messageContext) {
         return true;
     }
 
-    private JSONObject updateSystemLog(String logLevel) {
-
-        JSONObject jsonBody = new JSONObject();
-
-        // update root logger details
-        Logger rootLogger = Logger.getRootLogger();
-        rootLogger.setLevel(Level.toLevel(logLevel));
-
-        // update logger and appender data, following are set
-        Enumeration loggersEnum = LogManager.getCurrentLoggers();
-        Level systemLevel = Level.toLevel(logLevel);
-        while (loggersEnum.hasMoreElements()) {
-            Logger logger = (Logger) loggersEnum.nextElement();
-            logger.setLevel(systemLevel);
-        }
-        jsonBody.put(Constants.MESSAGE, "Successfully updated system log level to " + rootLogger.getLevel().toString());
-        return jsonBody;
-    }
+    private JSONObject updateLoggerData(org.apache.axis2.context.MessageContext axis2MessageContext, String loggerName,
+                                        String logLevel) {
 
-    private JSONObject updateLoggerData(org.apache.axis2.context.MessageContext axis2MessageContext, String loggerName, String loggerLevel) {
+        File log4j2PropertiesFile = new File(filePath);
+        String loggers = Utils.getProperty(log4j2PropertiesFile, "loggers");
 
-        JSONObject jsonBody = new JSONObject();
-        //update logger data in current system
-        Logger logger = LogManager.exists(loggerName);
+        PropertiesConfiguration config = new PropertiesConfiguration();
+        PropertiesConfigurationLayout layout = new PropertiesConfigurationLayout(config);
 
-        if (Objects.nonNull(logger)) {
-            logger.setLevel(Level.toLevel(loggerLevel));
-            jsonBody.put(Constants.MESSAGE, "Successfully updated log level of " + loggerName + " to " + logger.getLevel().toString());
-        } else {
-            jsonBody = Utils.createJsonErrorObject("Invalid logger " + loggerName);
+        try {
+            layout.load(new InputStreamReader(new FileInputStream(log4j2PropertiesFile)));
+            if (loggerName.equals(Constants.ROOT_LOGGER)) {
+                config.setProperty(loggerName + ".level", logLevel);
+                layout.save(new FileWriter(filePath, false));
+                Utils.updateLoggingConfiguration();
+                jsonBody.put(Constants.MESSAGE, "Successfully updated log level of rootLogger " + logLevel);
```

Comment:
Shall we extract these out to a new method? Can reduce duplication

Reply:
Done in 8d41bd6044fc5c95f39f6bf345f0876a1e3947d0